text/3488-deprecate-cargo-categories.md

```diff
@@ -0,0 +1,374 @@
+`maths` and `science/maths` terms being used.
+
+- **Refinement:** Nested hierarchies make it possible to refine the search space
+from e.g. `science` to `science/biology`. In a flat namespace you could
+potentially combine multiple keywords to achieve similar results. Displaying a
+list of [related keywords](https://github.com/rust-lang/crates.io/pull/488#issuecomment-265600285)
+to the one the user is currently looking at can help with this though.
+
+- **Clarity:** In a flat namespace the chance of a single keyword being used for
+multiple purposes is slightly increased. With nested hierarchies, the parent
+term can often give a bit more context on how the keyword/category is supposed
+to be used.
+
+- **Bike-shedding:** While flat namespaces already have bike-shedding potential,
+a nested namespace makes this worse by adding questions like: "should this
+name be nested under another name?" or "what parent name should this be nested
+under?"
+
+- **Implementation:** Nested categories on crates.io are currently implemented
+using the [`ltree` Postgres extension](https://www.postgresql.org/docs/current/ltree.html).
+This could be implemented differently as well, but from a purely technical
+view a flat list of terms is significantly easier to implement and maintain.
+Admittedly, parts of this are crates.io implementation details though.
+
+
+# Prior art
+
+Looking at other package repositories:
+
+- [PyPI](https://pypi.org) (Python) offers curated [classifiers](https://pypi.org/classifiers)
+without metadata. They don't seem to have an official policy on adding new
+classifiers, having instead taken the original list [from external sources](https://peps.python.org/pep-0301/#distutils-trove-classification).
+
+- [Hackage](https://hackage.haskell.org) (Haskell) has both keywords and
+uncurated categories without any metadata.
+
+- [NPM](https://www.npmjs.com) (JavaScript) has keywords.
+
+- [pub.dev](https://pub.dev) (Dart) has keywords under the name "topics".
+
+- [Maven Central](https://mvnrepository.com/repos/central) (Java),
+[NuGet](https://www.nuget.org) (.NET), [Packagist](https://packagist.org)
+(PHP) have keywords under the name "tags".
+
+- [Dub](https://code.dlang.org) (D), [Go](https://pkg.go.dev),
+[RubyGems](https://rubygems.org) (Ruby) have no keywords or categories.
+
+
+# Proposal / Motivation
+
+This RFC proposes to migrate crates.io from using categories **and** keywords
+to using only keywords, with a list of recommended keywords to address
+the duplication and clarity concerns from above. Keywords will keep their
+structure as a flat hierarchy, allowing crate owners to express nested
+categories can keyword combinations instead.
+
+An automated migration from categories to keywords for existing crates is
+currently not planned as the majority of crates that use categories also use
+keywords at the same time, while the reverse is not the case (see "Statistics"
+section above).
```

Comment:
This wording makes it seem like categories are going to still exist but be deprecated, whereas the reality is that the existing categories will be ignored instead. I think that this paragraph should say something about that.

Reply:
```suggestion
The plan is for crates.io to not show any of the category user interface elements
anymore. Eventually the crates.io server will also start to ignore the `categories`
field for new uploads and remove the existing categories data from the database.

An automated migration from categories to keywords for existing crates is
currently not planned as the majority of crates that have been using categories
are also using keywords at the same time, while the reverse is not the case
(see "Statistics" section above).
```

something like that?

Reply:
Looks good to me